tools/benchmark/cmd/watch.go

```diff
@@ -116,19 +124,22 @@ func watchFunc(cmd *cobra.Command, args []string) {
 
 	// put phase
 	// total number of puts * number of watchers on each key
-	eventsTotal := watchPutTotal * (watchTotal / watchedKeyTotal)
-
+	eventsTotal = watchPutTotal * (watchTotal / watchedKeyTotal)
 	results = make(chan result)
 	bar = pb.New(eventsTotal)
 
 	bar.Format("Bom !")
 	bar.Start()
 
+	for i := 0; i < len(streams); i++ {
+		putPhaseStartCh <- struct{}{}
+	}
+
 	putreqc := make(chan etcdserverpb.PutRequest)
 
+	wg.Add(1) // the doWatch() that received the last one watch event will trigger wg.Done()
```

Comment:
this is pretty tricky. I feel we should keep the old way. just leave the do watch routine there and count for the done of put routines.